exec/java-exec/src/main/java/org/apache/drill/exec/server/options/TypeValidators.java

```diff
@@ -204,10 +204,11 @@ public void validate(final OptionValue v, final OptionManager manager) {
    * Validator that checks if the given value is included in a list of acceptable values. Case insensitive.
    */
   public static class EnumeratedStringValidator extends StringValidator {
-    private final Set<String> valuesSet = new HashSet<>();
+    private final Set<String> valuesSet = new LinkedHashSet<>();
 
     public EnumeratedStringValidator(String name, String def, String... values) {
       super(name, def);
+      valuesSet.add(def.toLowerCase());
```

Comment:
instead of adding the default to the set, shouldn't we check that def is actually present in values? I think this is how this class has been used until now.

When checking the changes to `PlannerSettings.java`, it's impossible to know which value is actually the default one, whereas it becomes more obvious if the same value is present twice (not perfect though, but at least less error prone. A better change/validator would be to use enums and have something like: 
```
class EnumValidator<T extends Enum<T>> extends TypeValidator {
  public EnumValidator(String name, Enum<T> default, Class<T> enumClazz) {
    ...
  }
}
```
)